packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js

```diff
@@ -1629,6 +1633,176 @@ QUnit.module('Templates', () => {
     });
 });
 
+QUnit.module('Templates: fieldAddons', {
+    beforeEach: function() {
+        this.$container = $('#dropDownEditorLazy');
+
+        this.createEditor = function(options = {}) {
+            return this.$container
+                .dxDropDownEditor({
+                    value: 'test',
+                    fieldAddons: this.getDefaultAddons(),
+                    ...options,
+                })
+                .dxDropDownEditor('instance');
+        };
+
+        this.getDefaultAddons = function() {
+            return {
+                beforeTemplate: (value) => `before - ${value}`,
+                afterTemplate: (value) => `after - ${value}`,
+            };
+        };
+
+        this.getAddons = function() {
+            const $beforeAddon = this.$container.find(`.${DROP_DOWN_EDITOR_BEFORE_FIELD_ADDON}`);
+            const $afterAddon = this.$container.find(`.${DROP_DOWN_EDITOR_AFTER_FIELD_ADDON}`);
+
+            return {
+                beforeAddon: $beforeAddon.get(0),
+                afterAddon: $afterAddon.get(0),
+                $beforeAddon,
+                $afterAddon
+            };
+        };
+    },
+    afterEach: function() {
+        this.$container.remove();
+    }
+}, () => {
+    QUnit.test('should create addons addons once and reuse them on value change', function(assert) {
```

Comment:
Please check another tests as well
```suggestion
    QUnit.test('should create addons once and reuse them on value change', function(assert) {
```

Reply:
https://github.com/DevExpress/DevExtreme/pull/30766/commits/e37b0461cb0ba2dfade439a395c94feb29f3da2d